client/src/components/geneExpression/index.js

```diff
@@ -0,0 +1,183 @@
+import actions from "../../actions/";
+
+@connect(state => {
+  const metadata = _.get(state.controls.world, "obsAnnotations", null);
+  const ranges = _.get(state.controls.world, "summary.obs", null);
+  const initializeRanges = _.get(state.controls.world, "summary.obs");
+
+  return {
+    ranges,
+    metadata,
+    initializeRanges,
+    userDefinedGenes: state.userDefinedGenes,
+    world: state.controls.world,
+    colorAccessor: state.controls.colorAccessor,
+    allGeneNames: state.controls.allGeneNames
+  };
+})
+class GeneExpression extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      svg: null,
+      ctx: null,
+      axes: null,
+      dimensions: null,
+      gene: null
+    };
+  }
+
+  componentDidMount() {}
+  handleBrushAction(selection) {
+    this.props.dispatch({
+      type: "continuous selection using parallel coords brushing",
+      data: selection
+    });
+  }
+  handleColorAction(key) {
+    return () => {};
+    const indexOfGene = 0; /* we only get one, this comes from server as needed now */
+
+    const expressionMap = {};
+    /*
+      converts [{cellname: cell123, e}, ...]
+
+      expressionMap = {
+        cell123: [123, 2],
+        cell789: [0, 8]
+      }
+    */
+    // _.each(data.data.cells, cell => {
+    //   /* this action is coming directly from the server */
+    //   expressionMap[cell.cellname] = cell.e;
+    // });
+    //
+    // const minExpressionCell = _.minBy(data.data.cells, cell => {
+    //   return cell.e[indexOfGene];
+    // });
+    //
+    // const maxExpressionCell = _.maxBy(data.data.cells, cell => {
+    //   return cell.e[indexOfGene];
+    // });
+    // dispatch({
+    // type: "color by expression"
+    //   gene: gene,
+    //   colorAccessor: gene,
+    //   data,
+    //   expressionMap,
+    //   minExpressionCell,
+    //   maxExpressionCell,
+    //   indexOfGene
+    // });
+  }
+
+  render() {
+    return (
+      <div>
+        <input
+          onChange={e => {
+            this.setState({ gene: e.target.value });
+          }}
+          type="text"
+        />
+        <button
+          style={{
+            border: "none",
+            background: "none",
+            cursor: "pointer",
+            outline: "none",
+            marginTop: 20,
+            padding: 0
+          }}
+          onClick={() => {
+            this.props.dispatch(
+              actions.requestGeneExpressionCountsPOST([this.state.gene])
+            );
+          }}
+        >
+          <CirclePlus
+            style={{
+              display: "inline-block",
+              marginLeft: 7,
+              color: globals.brightBlue,
+              fontSize: 22
+            }}
+          />{" "}
+          add gene{" "}
+        </button>
+        {this.props.world
+          ? _.map(this.props.world.varDataCache, (value, key) => {
```

Comment:
As we discussed, I think a better approach is to:
* store a list of genes in state.control, eg, state.control.geneHistograms
* generate these histograms from that list
* anytime you add a gene to this list, also call the action handler to fetch the expression data and populate the varDataCache
* limit the size of this geneHistogram list, and ensure that the varDataCache lowWatermark config is larger than this limit
* make sure you keep state.controls.dimensionMap updated wiht any changes to this gene list, especially when world changes (see my notes on reducers/controls.js)